core/node/node_framework/src/implementations/layers/postgres_metrics.rs

```diff
@@ -0,0 +1,54 @@
+use std::time::Duration;
+
+use zksync_dal::{metrics::PostgresMetrics, ConnectionPool, Core};
+
+use crate::{
+    implementations::resources::pools::{PoolResource, ReplicaPool},
+    service::{ServiceContext, StopReceiver},
+    task::{Task, TaskId},
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+const SCRAPE_INTERVAL: Duration = Duration::from_secs(60);
+
+#[derive(Debug)]
+pub struct PostgresMetricsLayer;
+
+#[async_trait::async_trait]
+impl WiringLayer for PostgresMetricsLayer {
+    fn layer_name(&self) -> &'static str {
+        "postgres_metrics_layer"
+    }
+
+    async fn wire(self: Box<Self>, mut context: ServiceContext<'_>) -> Result<(), WiringError> {
+        let replica_pool_resource = context.get_resource::<PoolResource<ReplicaPool>>().await?;
+        let pool_for_metrics = replica_pool_resource.get_singleton().await?;
+        context.add_task(Box::new(PostgresMetricsScrapingTask { pool_for_metrics }));
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+struct PostgresMetricsScrapingTask {
+    pool_for_metrics: ConnectionPool<Core>,
+}
+
+#[async_trait::async_trait]
+impl Task for PostgresMetricsScrapingTask {
```

Comment:
`// Start scraping Postgres metrics before store initialisation as well.` comment in `core/bin/external_node/src/main.rs` makes me wonder why we  do `impl Task`  and not  `UnconstrainedTask`

Reply:
That's a good catch -- it should.

Reply:
I guess same for `PrometheusExporterLayer`.